sale_quotation_capable_vendor_rfq/models/sale_order.py

```diff
@@ -0,0 +1,175 @@
+            return self.action_view_rfq()
+        raise UserError(_("No RFQs created"))
+
+    def _get_capable_vendors(self, lead_time=0):
+        """
+        Returns a recordset of res.partner records that match the criteria
+
+        param lead_time: Integer of days to match
+
+        return: Recordset of res.partner
+        """
+        self.ensure_one()
+        vendors_dict = self._get_vendor_dict(lead_time)
+        return vendors_dict.keys()
+
+    def _create_rfq(self):
+        """
+        Create purchase orders from this sale order
+        """
+        self.ensure_one()
+        vendors_dict = self._get_vendor_dict()
+        for vendor, value in vendors_dict.items():
+            purchase = self.env["purchase.order"].create(self._prepare_rfq_vals(vendor))
+            for line in value:
+                self.env["purchase.order.line"].create(
+                    self._prepare_rfq_line_vals(purchase, line)
+                )
+        self._added_alternative_rfq()
+
+    def _prepare_rfq_vals(self, vendor):
+        """
+        Prepare values for purchase.order
+
+        :param vendor: res.partner
+
+        :return: dict
+        """
+        return {
+            "partner_id": vendor.id,
+            "origin": self.name,
+            "sale_quotation_id": self.id,
+            "currency_id": self.currency_id.id,
+            "company_id": self.company_id.id,
+        }
+
+    def _prepare_rfq_line_vals(self, rfq, line):
+        """
+        Prepare values for purchase.order.line
+
+        :param rfq: purchase.order
+        :param line: dict
+
+        :return: dict
+        """
+        return {
+            "order_id": rfq.id,
+            "product_id": line["line"].product_id.id,
+            "name": line["line"].name,
+            "product_qty": line["line"].product_uom_qty,
+            "product_uom": line["line"].product_uom.id,
+            "price_unit": self._convert_to_currency(
+                line["supplierinfo"].price,
+                line["supplierinfo"].currency_id,
+                rfq.currency_id,
+            ),
+            "date_planned": fields.Datetime.now(),
+        }
+
+    def _get_vendor_dict(self, lead_time=0):
+        """
+        Returns a dict of res.partner records that match the criteria
+
+        :param lead_time: Integer of days to match
+
+        :return: dict
+        """
+        vendors_dict = {}
+        for line in self.order_line:
+            supplierinfo_ids = line.product_id._get_matching_vendor_pricelists(
+                line.product_uom_qty, lead_time
+            )
+            for supplierinfo in supplierinfo_ids:
+                vals = {
+                    "supplierinfo": supplierinfo,
+                    "line": line,
+                }
+                if supplierinfo.partner_id not in vendors_dict:
+                    vendors_dict[supplierinfo.partner_id] = [vals]
+                else:
+                    vendors_dict[supplierinfo.partner_id].append(vals)
```

Comment:
You can use `defaultdict`.

`supplierinfo = defaultdict(list)`

```suggestion
                vendors_dict[supplierinfo.partner_id].append(vals)               
```